scipy/optimize/_differentialevolution.py

```diff
@@ -872,7 +872,7 @@ def _calculate_population_energies(self, population):
         try:
             calc_energies = list(self._mapwrapper(self.func,
                                                   parameters_pop[0:nfevs]))
-            energies[0:nfevs] = calc_energies
+            energies[0:nfevs] = np.squeeze(np.array(calc_energies))
```

Comment:
The intermediate use of `array` is not necessary.
```suggestion
            energies[0:nfevs] = np.squeeze(calc_energies)
```

-----

This is what I call an "indiscriminate squeeze" (i.e. `squeeze` with no axis argument), and those always make me nervous.  In this case, I think it is OK, but that is based on some assumptions that I haven't verified by digging into the code.

In particular, I assume this code triggers the new numpy issue when `self.func` returns a 1-d array with length 1 instead of a scalar.  Then, when converted to an array, `calc_energies` has shape (nfevs, 1), and assigning that to `energies[0:nfevs]` generates a warning with the development version of numpy.  E.g.

```
In [20]: z = np.empty(3)

In [21]: z[:] = [np.array([1]), np.array([2]), np.array([3])]                              
/Users/warren/mf37npdev/bin/ipython:1: DeprecationWarning: setting an array element with a sequence. This was supported in some cases where the elements are arrays with a single element. For example `np.array([1, np.array([2])], dtype=int)`. In the future this will raise the same ValueError as `np.array([1, [2]], dtype=int)`.
  #!/Users/warren/mf37npdev/bin/python
```

In released versions of numpy, that example works with no warning.

The potential problem with the indiscriminate squeeze is that *all* dimensions with length 1 are removed.  In this case, if `nfevs` is 1, `calc_energies` will be squeezed to a scalar array (i.e. an array with shape ()).  That shouldn't be a problem here, because it is OK to assign a scalar array into a 1-d array with length 1, e.g. this is not a problem:

```
In [22]: z[:1] = np.array(1)                                                               
```
So if my assumptions are correct, then this change (with my suggested simplification) looks fine.

Reply:
Your assumption is correct, I've just made a commit that removes the intermediate array call, and added an axis keyword.

Reply:
Looks like `calc_energies` has shape `(nfevs, 1)` only if `func` returns a 1-d array of length 1.  If `func` returns a true scalar, `calc_energies` has shape (nfevs,) and `squeeze` with `axis=1` doesn't work.

The code does the right thing without passing an `axis` to `squeeze`, so the argument could be removed. (An indiscriminate `squeeze` makes me worry, but if a careful review shows that it does the right thing, I'm fine with using it--in that case, it isn't really "indiscriminate".)